### PR TITLE
IAM Role delegation: files under buckets owned by a different AWS account

### PR DESCRIPTION
**Pull request by pegerto**
Hi, 

I need this functionality to use the s3 input, I am testing this and writing the unit testing.  

But I will appreciate feedback. 
Regards.

### CODE REVIEW

**Comment by pegerto** [comment]
Related to #50 , please review

File: spec/inputs/s3_spec.rb [hunk 1]
```diff
@@ -43,6 +43,23 @@
...
+        {
+          "access_key_id" => "1234",
+          "secret_access_key" => "secret",
+          "bucket" => "logstash-test",
+          "role_arn" => "arn:aws:iam::0000000:role/my_assumed_role"
+        }
+      }
+
+      it "assume new sts credential " do
+        subject.register
```
Comment (purbon) [inline spec/inputs/s3_spec.rb]: This test is not checking for the new credentials to be set properly, can you add some expectations to check that?

**Comment by purbon** [comment]
Thanks a lot for your contribution @pegerto , in order to move forward with your PR is going to be necessary to sign the CLA agreement, you can find more information at https://www.elastic.co/contributor-agreement.

**Comment by pegerto** [comment]
Hi Pere,

Thanks for your comments, I had signed the CLA agreement, hope its fine

**Comment by purbon** [comment]
Thanks for signing the CLA, on the other side as this adds a new functionality it would be nice if you could add test to cover it. Like this we could catch feature issues and regressions.

Thanks a lot!

- purbon

**Comment by purbon** [comment]
@ph your eyes more than welcome here :+1:

**Comment by purbon** [comment]
@pegerto I was checking your CLA signature and you did it with your personal email while in this PR has an email I guess from your employer (cisco). It would be required that you use the same email in both places the CLA and your PR, so we have two options here:

* Either sign the CLA with the email used in this PR.
* Change the author information in the commit.

/cheers

- purbon

**Comment by morganchristiansson** [comment]
Adds role_arn similarly to https://github.com/logstash-plugins/logstash-mixin-aws/pull/4/files

**Pull request closed without merging**